extensions/BugModal/web/comments.js

```diff
@@ -269,48 +269,54 @@ $(function() {
         var comment = that.parents('.comment');
         var commentNo = comment.data('no');
         var commentID = comment.data('id');
-        var tag = that.parent('.comment-tag').contents().filter(function() {
-            return this.nodeType === 3;
-        }).text();
-        var container = that.parents('.comment-tags');
+        var tagNode = that.parent('.comment-tag');
+        var deleteTag = tagNode.data('tag');
 
         // update ui
-        that.parent('.comment-tag').remove();
-        renderTags(commentNo, tagsFromDom(container));
+        tagNode.remove();
         updateTagsMenu();
 
         // update Bugzilla
         try {
-            renderTags(commentNo, await Bugzilla.API.put(`bug/comment/${commentID}/tags`, { remove: [tag] }));
+            var result = await Bugzilla.API.put(`bug/comment/${commentID}/tags`, { remove: [deleteTag] });
+            renderTags(commentNo, result.tags, result.tag_urls);
             updateTagsMenu();
         } catch ({ message }) {
             taggingError(commentNo, message);
         }
     }
-    $('.comment-tag a').click(deleteTag);
+    $('.comment-tag a.remove').click(deleteTag);
 
     function tagsFromDom(commentTagsDiv) {
         return commentTagsDiv
             .find('.comment-tag')
-            .contents()
-            .filter(function() { return this.nodeType === 3; })
-            .map(function() { return $(this).text(); })
+            .map(function() {return $(this).data('tag');})
             .toArray();
     }
 
-    function renderTags(commentNo, tags) {
+    function renderTags(commentNo, tags, tag_urls) {
         cancelRefresh();
         var root = $('#ctag-' + commentNo + ' .comment-tags');
         root.find('.comment-tag').remove();
         $.each(tags, function() {
-            var span = $('<span/>').addClass('comment-tag').text(this);
+            var span = $("<span/>")
+                .addClass("comment-tag")
+                .data("tag", this)
+                .text(this);
             if (BUGZILLA.user.can_tag) {
                 span.prepend($('<a role="button" aria-label="Remove" class="remove">×</a>').click(deleteTag));
             }
+            if (tag_urls && tag_urls[this]) {
+                var infoLink = $('<a aria-label="More Information" target="_blank"></a>');
+                infoLink.attr('href', tag_urls[this].htmlEncode());
+                var img = $('<img width="16" height="16" title="Click for more information"/>');
+                img.attr('src', BUGZILLA.config.basepath + 'extensions/BugModal/web/images/help.png');
+                infoLink.append(img);
+                span.append(infoLink);
+            }
```

Comment:
Rather than duplicating rendering both server and client side, would it be better to extract the comment tags template into its own template, then always request a html fragment from the client?